cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceQueryAdmin.java

```diff
@@ -103,11 +102,6 @@ public NamespaceMeta get(Id.Namespace namespaceId) throws Exception {
    */
   @Override
   public boolean exists(Id.Namespace namespaceId) throws Exception {
-    try {
-      get(namespaceId);
-    } catch (NotFoundException e) {
-      return false;
-    }
-    return true;
+    return nsStore.get(namespaceId) != null;
```

Comment:
That's ok. In that case, we can use the response code of `get()` to implement exists. If it is 404, then it is `false`, otherwise if 200 or unauthorized, then it is true.